infra/tests/tests_views.py

```diff
@@ -0,0 +1,140 @@
+        self.assertEqual(infra.models.OperationLog.objects.get().referrer,
+                         referrer)
+        self.assertEqual(infra.models.OperationLog.objects.get().user_agent,
+                         user_agent)
+
+    def test_list_operationlog(self):
+        '''operationlog list should be accessed by GET request.'''
+        url = reverse('operationlog-list')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_delete_operationlog(self):
+        '''operationlog should be deleted by DELETE request.'''
+        department = mommy.make(infra.models.OperationLog)
+        url = reverse('operationlog-detail', args=(department.pk,))
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(infra.models.OperationLog.objects.count(), 0)
+
+    def test_get_operationlog(self):
+        '''operationlog should be accessed by GET request.'''
+        department = mommy.make(infra.models.OperationLog)
+        url = reverse('operationlog-detail', args=(department.pk,))
+        expected_keys = {'id', 'time', 'remote_ip', 'requester', 'method',
+                         'url', 'referrer', 'user_agent'}
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(set(response.data.keys()), expected_keys)
+
+    def test_update_operationlog(self):
+        '''Operationlog should be updated by PATCH request.'''
+        method0 = 1
+        method1 = 2
+        department = mommy.make(infra.models.OperationLog, method=method0)
+        url = reverse('operationlog-detail', args=(department.pk,))
+        data = {'method': method1}
+
+        response = self.client.patch(url, data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn('method', response.data)
+        self.assertEqual(response.data['method'], method1)
+
+
+class TestNotificationViewSet(APITestCase):
+    '''Unit tests for Notification view.'''
+    def test_create_notification(self):
+        '''Notification should be created by POST request.'''
+        # notification = mommy.make(infra.models.Notification)
+        url = reverse('notification-list')
+        user0 = mommy.make(User)
+        sender = user0.id
+        user1 = mommy.make(User)
+        recipient = user1.id
+        content = "test"
+        read_time = now()
+        data = {'sender': sender, 'recipient': recipient,
+                'content': content, 'read_time': read_time}
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(infra.models.Notification.objects.count(), 1)
+        self.assertEqual(infra.models.Notification.objects.get().read_time,
```

Comment:
Hmm, while this assertion is OK, we could test the `sender` and the `recipient`. When the notification is created, it's not read, we should update the `read_time` when `recipient` read it, but this could be tested in another test.